datafusion/physical-expr/src/window/nth_value.rs

```diff
@@ -210,9 +225,31 @@ impl PartitionEvaluator for NthValueEvaluator {
                 // We produce None if the window is empty.
                 return ScalarValue::try_from(arr.data_type());
             }
+
+            let mut valid_indices = Vec::new();
+            if self.ignore_nulls {
+                valid_indices = arr.nulls().unwrap().valid_indices().collect::<Vec<_>>();
```

Comment:
I think, the unexpected result I mentioned is related to here. You might need calculate valid_indices within the `range` not on whole `arr`.